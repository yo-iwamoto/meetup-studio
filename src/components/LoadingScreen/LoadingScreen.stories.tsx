import { LoadingScreen } from '.';
import type { Meta, StoryObj } from '@storybook/react';

const meta = {
  component: LoadingScreen,
} satisfies Meta<typeof LoadingScreen>;

export default meta;

export const Default = {} satisfies StoryObj<typeof meta>;
