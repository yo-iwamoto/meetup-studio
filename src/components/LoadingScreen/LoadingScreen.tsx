export const LoadingScreen = () => {
  return (
    <div className='fixed inset-0 flex flex-col items-center justify-center gap-8'>
      <div className='h-16 w-16 animate-spin-slow rounded-xl bg-orange-300'></div>
      <p className='text-2xl text-orange-400'>loading...</p>
    </div>
  );
};
