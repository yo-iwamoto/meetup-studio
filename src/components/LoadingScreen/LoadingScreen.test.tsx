import * as stories from './LoadingScreen.stories';
import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';

const { Default } = composeStories(stories);

describe('LoadingScreen', () => {
  it('renders unchanged', () => {
    const { getByText, asFragment } = render(<Default />);

    expect(getByText('loading...')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
