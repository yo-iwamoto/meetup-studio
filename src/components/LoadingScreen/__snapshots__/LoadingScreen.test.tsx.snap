// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingScreen > renders unchanged 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 flex flex-col items-center justify-center gap-8"
  >
    <div
      class="h-16 w-16 animate-spin-slow rounded-xl bg-orange-300"
    />
    <p
      class="text-2xl text-orange-400"
    >
      loading...
    </p>
  </div>
</DocumentFragment>
`;
